cmd/operator-sdk/bundle/validate.go

```diff
@@ -190,16 +208,34 @@ func (c bundleValidateCmd) run() (res internal.Result) {
 	// from the ValidateBundleContent to add the output(s) into the result
 	checkResults(results, &res)
 
-	return res
+	return res, nil
+}
+
+// newImageRegistryForTool returns an image registry based on what type of image tool is passed.
+// If toolStr is empty, a containerd registry is returned.
+func newImageRegistryForTool(logger *log.Entry, toolStr string) (reg registryimage.Registry, err error) {
```

Comment:
it is not working. See;

- It is allowing invalid values as arg of `--image-builder=invalid`

``` $ operator-sdk bundle validate ./deploy/olm-catalog/memcached-operator/ --image-builder=invalid
INFO[0000] Found annotations file                        bundle-dir=deploy/olm-catalog/memcached-operator container-tool=invalid
INFO[0000] Could not find optional dependencies file     bundle-dir=deploy/olm-catalog/memcached-operator container-tool=invalid
INFO[0000] All validation tests have completed successfully 
```

Reply:
Also, see: 
```

 $ operator-sdk bundle validate quay.io/example/test-operator:v0.1.0
INFO[0000] Unpacking image layers                       
INFO[0000] running docker create                        
INFO[0000] running docker cp                            
INFO[0000] running docker rm                            
INFO[0000] Found annotations file                        bundle-dir=/var/folders/r_/ym6yn3cd0q3fm5jcgcwvj4z00000gn/T/bundle-154915825 container-tool=docker
INFO[0000] Could not find optional dependencies file     bundle-dir=/var/folders/r_/ym6yn3cd0q3fm5jcgcwvj4z00000gn/T/bundle-154915825 container-tool=docker
INFO[0000] All validation tests have completed successfully 
camilamacedo@Camilas-MacBook-Pro ~/go/src/github.com/operator-framework/operator-sdk-samples/go/memcached-operator (test-go) $ operator-sdk bundle validate quay.io/example/test-operator:v0.1.0 --image-builder=invalid
INFO[0000] Unpacking image layers                       
INFO[0000] All validation tests have completed successfully 
camilamacedo@Camilas-MacBook-Pro ~/go/src/github.com/operator-framework/operator-sdk-samples/go/memcached-operator (test-go) $ 
```

Reply:
This "problem" comes from how operator-registry creates a `ContainerTool` type, which doesn't actually validate the passed string. Instead it chooses a default, which is not the correct behavior for a CLI. Updated.

Reply:
I think we can validate the string informed in the cmd since we set the possible values in the flag.

Reply:
Already fixed.